leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/observe/ObserveCompositeTest.java

```diff
@@ -0,0 +1,253 @@
+ * are made available under the terms of the Eclipse Public License v2.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v20.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *     Michał Wadowski (Orange) - Add Observe-Composite feature.
+ *******************************************************************************/
+package org.eclipse.leshan.integration.tests.observe;
+
+import org.eclipse.californium.core.coap.Response;
+import org.eclipse.leshan.core.ResponseCode;
+import org.eclipse.leshan.core.node.LwM2mNode;
+import org.eclipse.leshan.core.node.LwM2mPath;
+import org.eclipse.leshan.core.node.LwM2mSingleResource;
+import org.eclipse.leshan.core.observation.Observation;
+import org.eclipse.leshan.core.observation.CompositeObservation;
+import org.eclipse.leshan.core.request.*;
+import org.eclipse.leshan.core.response.LwM2mResponse;
+import org.eclipse.leshan.core.response.ObserveCompositeResponse;
+import org.eclipse.leshan.core.response.ReadResponse;
+import org.eclipse.leshan.core.response.WriteCompositeResponse;
+import org.eclipse.leshan.integration.tests.util.IntegrationTestHelper;
+import org.eclipse.leshan.server.registration.Registration;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsInstanceOf.instanceOf;
+import static org.junit.Assert.*;
+
+public class ObserveCompositeTest {
+
+    private final String examplePath1 = "/3/0/15";
+    private final String examplePath2 = "/3/0/14";
+    private final String exampleValue1 = "Europe/Paris";
+    private final String exampleValue2 = "+11";
+
+    protected IntegrationTestHelper helper = new IntegrationTestHelper();
+    private Registration currentRegistration;
+    private TestObservationListener listener;
+
+    @Before
+    public void start() {
+        helper.initialize();
+        helper.createServer();
+        helper.server.start();
+        helper.createClient();
+        helper.client.start();
+        helper.waitForRegistrationAtServerSide(1);
+
+        currentRegistration = helper.getCurrentRegistration();
+        listener = new TestObservationListener();
+        helper.server.getObservationService().addListener(listener);
+    }
+
+    @After
+    public void stop() {
+        helper.client.destroy(false);
+        helper.server.destroy();
+        helper.dispose();
+    }
+
+    @Test
+    public void can_composite_observe_on_single_resource() throws InterruptedException {
+        ObserveCompositeResponse observeResponse = sendObserveCompose(examplePath1);
+
+        assertObserveCompositeResponse(observeResponse);
```

Comment:
assertObserveCompositeResponse is not so clear what is asserted ? 
maybe assertIsValidObserveCompositeResponse ?